v3/newrelic/transaction.go

```diff
@@ -324,7 +333,7 @@ func (txn *Transaction) startSegmentAt(at time.Time) SegmentStartTime {
 //	// ... code you want to time here ...
 //	segment.End()
 func (txn *Transaction) StartSegment(name string) *Segment {
-	if IsSecurityAgentPresent() && txn != nil && txn.thread != nil && txn.thread.thread != nil && txn.thread.thread.threadID > 0 {
+	if IsSecurityAgentPresent() && nilTransaction(txn) && txn.thread.thread.threadID > 0 {
 		// async segment start
```

Comment:
Shouldn't this be `!nilTransaction(txn)`?

Reply:
Also, `nilTransaction(txn)` doesn't test for `txn.thread.thread != nil`

Reply:
didnt know that was possible 😮 
--> will update

Reply:
how does this look?